website_product_subscription/controllers/subscribe.py

```diff
@@ -0,0 +1,268 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Coop IT Easy SCRLfs <http://coopiteasy.be>
+#   Rémy Taymans <remy@coopiteasy.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import http
+from openerp.exceptions import ValidationError
+from openerp.http import request
+from openerp.tools.translate import _
+
+from openerp.addons.website_product_subscription.controllers.subscribe_form import SubscribeForm
+
+
+class SubscribeController(http.Controller):
+
+    @http.route(
+        [
+            # '/page/become_subscriber',
+            # '/become_subscriber',
+            '/new/subscription',
+        ],
+        type='http',
+        auth='public',
+        website=True
+    )
+    def subscribe(self, **kwargs):
+        form = self.subscribe_form
+        form.normalize_form_data()
+        form.validate_form()
+        form.init_form_data()
+        self.fill_values(request.params)
+        form.set_form_defaults()
+        if ('error' not in request.params
+                and request.httprequest.method == 'POST'):
+            sub_req = self.process_subscribe_form(form)
+            values = {
+                'subscription_request_id': sub_req,
+                'subscriber': sub_req.subscriber.id,
+                'subscription_template': sub_req.subscription_template.id,
+                'gift': 'off',
+                'sponsor': sub_req.sponsor.id if sub_req.sponsor else '',
+            }
+            if 'redirect' in kwargs:
+                return request.redirect(kwargs.get('redirect'))
+            else:
+                return self.get_subscription_response(values, kwargs)
+        return request.website.render(
+           'website_product_subscription.subscribe_form', request.params
+        )
+
+    @http.route(
+        '/new/gift/subscription',
```

Comment:
@remytms  for  consistency, it should be
- `/new/subscription/gitf`
and the other 
- `/new/subscription/basic` 

or even `susbcription/create` (ressource than REST verb)

Reply:
@robinkeunen Just THANKS ! I will reword URL this way. :D